discord/primary_guild.py

```diff
@@ -0,0 +1,91 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-present Rapptz
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from typing import Optional, TYPE_CHECKING
+from datetime import datetime
+
+from .asset import Asset
+from .utils import snowflake_time
+
+if TYPE_CHECKING:
+    from .state import ConnectionState
+    from .types.primary_guild import PrimaryGuild as PrimaryGuildPayload
+
+
+class PrimaryGuild:
+    r"""Represents the primary guild (formally known as a clan) of a :class:`User`"""
+    __slots__ = ('_guild_id', 'identity_enabled', '_tag', '_badge', '_state')
+
+    if TYPE_CHECKING:
+        _guild_id: int
+        identity_enabled: bool
+        _tag: str
+        _badge: str
+        _state: ConnectionState
+
+    def __init__(self, *, state, data: PrimaryGuildPayload) -> None:
+        self._state = state
+        self._update(data)
+
+    def _update(self, data: PrimaryGuildPayload):
+        self._guild_id = data.get('identity_guild_id', None)
```

Comment:
```suggestion
        self.id = _get_as_snowflake(data, 'identity_guild_id')
```
i feel like `user.primary_guild.guild_id` is a little redundant, and not much point to the property (also make sure to add the utils import :p)